proto/iotics/api/meta.proto

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2019-2020 Iotic Labs Ltd. All rights reserved.
+
+// Iotics Web protocol definitions (meta)
+syntax = "proto3";
+
+import "google/rpc/status.proto";
+import "iotics/api/common.proto";
+
+package iotics.api;
+
+option csharp_namespace = "Iotics.Api";
+option go_package = "github.com/Iotic-Labs/iotic-go-proto-qapi/iotics/api;ioticsapi";
+option java_multiple_files = true;
+option java_outer_classname = "MetaProto";
+option java_package = "com.iotics.api";
+option objc_class_prefix = "IAX";
+option php_namespace = "Iotics\\Api";
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+
+// MetaAPI enables querying of metadata associated with Twins and Feeds.
+service MetaAPI {
+
+  // SparqlQuery performs a SPARQL 1.1 query and returns one or more results, each as a sequence of chunks. Note that:
+  // - Chunks for a particular result will arrive in-order though they might be interleaved with chunks from other
+  //   results (when performing a non-local query). See scope parameter in SparqlQueryRequest.
+  // - The call will only complete once the (specified or host default) request timeout has been reached. The client can
+  //   choose to end the stream early once they have received enough results. (E.g. in the case of Scope.LOCAL this
+  //   would be after the one and only sequence of chunks has been received.)
+  rpc SparqlQuery(SparqlQueryRequest) returns (stream SparqlQueryResultChunk) {}
+}
+
+// Available result content types for SPARQL requests. Note that applicable content types depend on the type of query.
+// Specifying an incompatible result type can lead to the result being returned with a different content type.
+enum SparqlResultType {
+
+  // Applicable to SELECT/ASK (SPARQL 1.1 Query Results JSON Format)
+  SPARQL_JSON = 0;
+  // Applicable to SELECT/ASK (SPARQL 1.1 Query Results XML Format)
+  SPARQL_XML = 1;
+  // Applicable to SELECT/ASK (SPARQL 1.1. Query Results CSV Format)
+  SPARQL_CSV = 2;
+  // Applicable to CONSTRUCT/DESCRIBE (Terse RDF Triple Language)
+  RDF_TURTLE = 3;
+  // Applicable to CONSTRUCT/DESCRIBE (RDF 1.1 XML)
+  RDF_XML = 4;
+  // Applicable to CONSTRUCT/DESCRIBE (RDF 1.1 N-Triples)
+  RDF_NTRIPLES = 5;
+  // Applicable to CONSTRUCT/DESCRIBE (RDF 1.1 N-Quads)
+  RDF_NQUADS = 6;
+}
+
+// SparqlQueryRequest describes a SPARQL query.
+message SparqlQueryRequest {
+
+  // SPARQL request payload.
+  message Payload {
+
+    // The desired result content type. Note that the actual content (mime) type could be different and will be returned
+    // in result chunks.
+    SparqlResultType resultTypeHint = 1;
```

Comment:
resultContentType ?

Reply:
I had exactly that before - I was wondering whether _hint_ is more appropriate because this is after all an `Accept:` and as per note against the enum, it's not guaranteed to that.
However, `resultContentType` would also be OK.

Reply:
definitely want to update this if thinking of failing fast server side on an incorrect type/query mapping

Reply:
FYI - "failing fast" will still mean running the query until response headers are received from SPARQL endpoint and `Accept` can be compared to `Content-Type`.

Reply:
e98aee517d91177a874f70b8cdec0821d66a7c75